### ISSUE TITLE
Panic

### ISSUE THREAD

**Issue by sftse** [jugglerchris/rust-html2text#164]
In commit 5b9d69f a regression was introduced, since then this fragment panics
```
<html>
    <body>
        <ul>
            <table><tr></tr></table>
        </ul>
    </body>
</html>
```

**Comment by jugglerchris** [comment]
Thanks for the reports and PRs!  I just wanted to get back to you to say that I will get to them, hopefully in the next week or two - life is busy right now and other things are a higher priority in the next few days!

**Comment by sftse** [comment]
I was surprised at the fact that this code does not crash with `<ol>` tags, upon looking closely [ol handling filters out some nodes that il handling does not](https://github.com/jugglerchris/rust-html2text/blob/0476d71633ce93c4da15638a27f0091df44bbf50/src/lib.rs#L1313-L1316)

Can you add a comment explaining the asymmetry, i.e. why lists seem to have non-`RenderNodeInfo::ListItem` inside them that must be cleaned to not throw off list numbering?

**Comment by jugglerchris** [comment]
I've merged #165  MR - thanks!  Leaving this open so that I can address https://github.com/jugglerchris/rust-html2text/issues/164#issuecomment-2352712814 when I have time.
Thanks again!
